Add 16A8W support and test for add operation

Ninja91 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #13653

Add 16A8W quantization support and test for the add operation in ExecutorTorch ARM backend.

This follows the pattern established for linear operations, extending int16 support to add operations.

Changes:
- Add INT16 dtype validation support in op_add.py
- Add test_add_tensor_16a8w_tosa_INT test function
- Enable test_add.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510463](https://our.internmc.facebook.com/intern/diff/D80510463/)

Ninja91 committed:
Add 16A8W support and test for add operation

Add 16A8W quantization support and test for the add operation in ExecutorTorch ARM backend.

This follows the pattern established for linear operations, extending int16 support to add operations.

Changes:
- Add INT16 dtype validation support in op_add.py
- Add test_add_tensor_16a8w_tosa_INT test function
- Enable test_add.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510463](https://our.internmc.facebook.com/intern/diff/D80510463/)

[ghstack-poisoned]

Ninja91 committed:
Update on "Add 16A8W support and test for add operation"


Add 16A8W quantization support and test for the add operation in ExecutorTorch ARM backend.

This follows the pattern established for linear operations, extending int16 support to add operations.

Changes:
- Add INT16 dtype validation support in op_add.py
- Add test_add_tensor_16a8w_tosa_INT test function
- Enable test_add.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510463](https://our.internmc.facebook.com/intern/diff/D80510463/)

[ghstack-poisoned]

digantdesai commented on backends/arm/test/ops/test_add.py:
```diff
@@ -216,3 +220,46 @@ def test_add_tensor_vgf_INT(test_data: input_t1):
+
+
+@common.parametrize("test_data", Add.test_data)
+def test_add_tensor_16a8w_tosa_INT(test_data: input_t1):
```
mark xfail and create a ticket?

```
Reference model version 1.0.0d is backward compatible with serializer version 1.0.0.
[0:GT]: Creating operator id_000,      ADD, 2 input tensors, 1 output tensors
SubgraphTraverser::initializeGraph(): OpFactory could not allocate op      ADD input=(INT16 rank 1) -> (INT16 rank 1)SubgraphTraverser::initializeGraph(): Input: quantized_decomposed_quantize_per_tensor_default
SubgraphTraverser::initializeGraph(): Input: quantized_decomposed_quantize_per_tensor_default
SubgraphTraverser::initializeGraph(): Output: aten_add_tensor
SUBGRAPH_ERROR_IF() fails AT /tmp/pip-install-_y2vr3od/tosa-tools_a7906fc1445b465b8c8bb06a1be1a528/reference_model/src/subgraph_traverser.cpp:402 initializeGraph(): (true)
SubgraphTraverser::initializeGraph(): Unsupported operation type or rank.
Caught signal 6 (Aborted)
```

digantdesai reviewed (changes requested):
LGTM, mark xfail and we can land this.

Ninja91 committed:
Update on "Add 16A8W support and test for add operation"


Add 16A8W quantization support and test for the add operation in ExecutorTorch ARM backend.

This follows the pattern established for linear operations, extending int16 support to add operations.

Changes:
- Add INT16 dtype validation support in op_add.py
- Add test_add_tensor_16a8w_tosa_INT test function
- Enable test_add.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510463](https://our.internmc.facebook.com/intern/diff/D80510463/)

[ghstack-poisoned]

Ninja91 committed:
Update on "Add 16A8W support and test for add operation"


Add 16A8W quantization support and test for the add operation in ExecutorTorch ARM backend.

This follows the pattern established for linear operations, extending int16 support to add operations.

Changes:
- Add INT16 dtype validation support in op_add.py
- Add test_add_tensor_16a8w_tosa_INT test function
- Enable test_add.py in test targets configuration

The 16A8W configuration uses 16-bit activations with 8-bit weights, enabling higher precision for activations while maintaining weight efficiency.

Differential Revision: [D80510463](https://our.internmc.facebook.com/intern/diff/D80510463/)

[ghstack-poisoned]
